src/plugins/intel_cpu/src/nodes/experimental_detectron_generate_proposals_single_image.cpp

```diff
@@ -411,8 +287,28 @@ void ExperimentalDetectronGenerateProposalsSingleImage::execute(dnnl::stream str
                               });
 
             unpack_boxes(reinterpret_cast<float *>(&proposals_[0]), &unpacked_boxes[0], pre_nms_topn);
-            nms_cpu(pre_nms_topn, &is_dead[0], &unpacked_boxes[0], &roi_indices_[0], &num_rois, 0,
+            if (n > 0)
+                std::fill(is_dead.begin(), is_dead.end(), 0);
+#ifdef __GNUC__
```

Comment:
Could you please clarify why we need conditional compilation for GNU here?

Reply:
There is a branch prediction hint support in gcc and clang compilers described in [6.59](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html) while MSVC doesn't support such option. Very similar feature is supported in C++20 through `[[likely]]` keyword, but we cannot use it since CPU plugin is compiled with C++11 option.